test/TestDestroy.cpp

```diff
@@ -27,17 +27,27 @@ TEST(TestDestroy, TestDestroyTensorSingle)
         {
             kp::Manager mgr;
 
-            tensorA = mgr.tensor({ 0, 0, 0 });
+            const std::vector<float> initialValues = {0.0f, 0.0f, 0.0f};
+
+            tensorA = mgr.tensor(initialValues);
 
             std::shared_ptr<kp::Algorithm> algo =
-              mgr.algorithm({ tensorA }, spirv);
+              mgr.algorithm({tensorA}, spirv);
+
+            // Sync values to and from device
+            mgr.sequence()
+            ->eval<kp::OpTensorSyncDevice>(algo->getTensors())
+            ->eval<kp::OpTensorSyncLocal>(algo->getTensors());
+
+            EXPECT_EQ(tensorA->vector(), initialValues);
 
             mgr.sequence()
               ->record<kp::OpAlgoDispatch>(algo)
```

Comment:
Maybe I should have asked this before creating the PR, but why doesn't the `record` command sync the tensors that are referenced by the `algo` to the device? Is it always necessary to sync them in client code with `OpTensorSyncDevice`?

Reply:
In this case you don't really need the "OpTensorSyncLocal". The SyncDevice basically copies tha information from your RAM to your vRAM, which is what you'll want to do always when you want to properly configure the GPU memory. The SyncLocal copies the value from vRAM to RAM, which is what you want once you want to retrieve the values from the GPU back to the "CPU visible RAM"